src/proxy/admin.rs

```diff
@@ -29,25 +29,22 @@ pub struct Admin {
     log: Logger,
     /// The address that the Admin server starts on
     addr: SocketAddr,
-    metrics: Arc<Metrics>,
     health: Arc<Health>,
 }
 
 impl Admin {
-    pub fn new(base: &Logger, addr: SocketAddr, metrics: Arc<Metrics>, heath: Health) -> Self {
-        Admin {
+    pub fn new(base: &Logger, addr: SocketAddr, heath: Health) -> Self {
+        Self {
             log: base.new(o!("source" => "proxy::Admin")),
             addr,
-            metrics,
             health: Arc::new(heath),
         }
     }
 
-    pub fn run(&self, mut shutdown_rx: watch::Receiver<()>) {
+    pub fn spawn(self, metrics: Arc<Metrics>, mut shutdown_rx: watch::Receiver<()>) {
```

Comment:
I know these changes happen mid-thought as we are working on a feature, but can we please avoid slipping in these kind of renames and breaks with current conventions that aren't tied to the current PR? (and this is a large PR at that, so it's easy for them to pass through without consensus being formed/proper review). It feels like this is becoming a bit of a habit.

If we have reason to change conventions, then that's fine - but can we tackle that in its own PR/ticket and draw consensus for it. At the very, very least, let's call it out in a PR (if it's small/medium sized PR), so that it can get the appropriate review.